### ISSUE TITLE
for in loop on object should not compile

### ISSUE THREAD

**Issue by familyfriendlymikey** [imba/imba#634]
This code doesn't print anything, but it neither throws at compile time nor runtime.
I think it makes sense that this shouldn't compile.
```
let s = new Set()
s.add(5)

for x in s
    console.log x
```

**Comment by somebee** [comment]
Yeah, this is the challenge with the `for x in` and `for x of`. We could, in development-mode identify when you try to loop through a set this way and actually warn in console that you have to use `for of`. Also, the way `for in` works is that it looks for and potentially calls `toIterable` on the object to loop. This is undocumented because we are likely to change the method name to a symbol to avoid conflicts. But to make sets iterable this way you could do
```imba
extend class Set
    def toIterable
        Array.from(self)
```
